### PR TITLE
change FileInfo to interface

### PR DESCRIPTION
**Pull request by ztplz**

### CODE REVIEW

**Comment by ztplz** [comment]
#732

File: js/stat.ts [hunk 1]
```diff
@@ -38,6 +37,34 @@ export class FileInfo {
...
+  isDirectory(): boolean;
+
+  /**
+   * Returns whether this is info for a symlink. This result is
+   * mutually exclusive to `FileInfo.isFile` and `FileInfo.isDirectory`.
+   */
+  isSymlink(): boolean;
+}
+
+class DenoFileInfo implements FileInfo {
```
Comment (ry) [inline js/stat.ts]: Call it FileInfoImpl please

Comment (ztplz) [reply js/stat.ts]: @ry  It is ok.

**Comment by ztplz** [comment]
@ry   I build success in my travis and appveyor, but I don't know why failed there.

**Pull request closed without merging**